cmd/kube-proxy/app/options.go

```diff
@@ -328,6 +315,50 @@ func (o *Options) processV1Alpha1Flags(fs *pflag.FlagSet) {
 	if fs.Changed("cluster-cidr") {
 		o.config.DetectLocal.ClusterCIDRs = strings.Split(o.clusterCIDRs, ",")
 	}
+	if fs.Changed("healthz-bind-address") {
+		host, port, err := net.SplitHostPort(o.healthzBindAddress)
+		var ip net.IP
+
+		// o.healthzBindAddress can have both "ip" and "ip:port" as value,
+		// we need to handle both cases.
+		if err != nil {
+			ip = netutils.ParseIPSloppy(o.healthzBindAddress)
+		} else {
+			ip = netutils.ParseIPSloppy(host)
+			intPort, _ := strconv.Atoi(port)
+			o.config.HealthzBindPort = int32(intPort)
```

Comment:
we validate in the config that the port exist and is valid

https://github.com/kubernetes/kubernetes/blob/d001d5684e69b08e120dc328917925318c5f324c/pkg/proxy/apis/config/validation/validation.go#L77-L80

https://github.com/kubernetes/kubernetes/blob/d001d5684e69b08e120dc328917925318c5f324c/pkg/proxy/apis/config/validation/validation.go#L225-L244

why should we not do that here?